phoenix-core/src/test/java/org/apache/phoenix/expression/InListExpressionTest.java

```diff
@@ -62,4 +68,83 @@ public void testHashCode() throws Exception {
                     + hashCode, firstHashCode, hashCode);
         }
     }
+
+    @Test
+    public void testKeyPartsForOrderMatters() {
```

Comment:
This is not a great unit test as its testing multiple things.  Consider reducing the scope and adding additional tests.  Same for below.

Reply:
@dbwong this is only testing one thing here, which is whether we have pk ordering or not.
Due to the way how Phoenix implemented, it's very hard to do the unit tests.

@ChinmaySKulkarni sure will do.